src/intermediate_p2.py

```diff
@@ -0,0 +1,280 @@
+
+from collections import UserList
+import random
+
+
+
+
+class Deck(UserList) :
+
+    def __init__(self):
+        UserList.__init__(self)
+
+
+    def add_card(self, card):
+        # check if card already exists?
+        if card in self:
+            print('Card already in deck.')
+        else:
+            self.append(card)
+
+
+    def remove_card(self, card):
+        print("in remove")
+        if card in self:
+            self.remove(card)
+        else:
+            print("Card cannot be removed. Card not present in deck")
+
+
+
+class HandDeck(Deck):
+    def __init__(self):
+        Deck.__init__(self)
+
+
+    def add_card(self, card):
+        if len(self) >= 2 :
+            print("HandDeck can have max. two cards")
+        else :
+            super(HandDeck, self).add_card(card)
+
+
+
+class DealerDeck(Deck):
+    colors = ["SPADES", "HEARTS", "DIAMONDS", "CLUBS"]
+
+    def __init__(self):
+        Deck.__init__(self)
+
+        for color in self.colors:
+            for num in range(2, 16):
+                if num == 11:
+                    continue
+                new_card = Card(num, color)
+                self.add_card(new_card)
+
+
+    def shuffle_cards(self):
+        middle = int(len(self) / 2)
+        num_of_cards = len(self)
+
+        for i in range(1, middle):
+            first_random_card_position = random.randint(1, middle)
+            second_random_card_position = random.randint(middle + 1, num_of_cards - 1)
+            temp = self[first_random_card_position]
+            self[first_random_card_position] = self[second_random_card_position]
+            self[second_random_card_position] = temp
+
+
+
+    def sort_cards_by_number(self, card):
+        def sortFirst(card):
+            return card.card_number
+
+        self.sort(key=sortFirst)
+
+
+
+
+    def sort_cards_by_color(self, card):
```

Comment:
Do you think you could merge these 2 functions in 1 sort_cards? So it would sort firstly by colour and then by number?
Hint: http://www.lleess.com/2013/08/python-sort-list-by-multiple-attributes.html#.XTG-afkzZTY

Reply:
Done